rskj-core/src/main/java/co/rsk/scoring/PeerScoringManager.java

```diff
@@ -241,18 +258,37 @@ public PeerScoring getPeerScoring(InetAddress address) {
     }
 
     /**
-     * Calculates the reputation for a peer
-     * Starts punishment if needed
-     *
-     * @param scoring       the peer scoring
-     * @param calculator    the calculator to use
+     * Records an event and starts punishment if needed
+     * @param peerScoring the peer scoring
+     * @param event an event type
+     * @param punishmentCalculator calculator to use
+     * @param nodeID a node id
      */
-    private void recordEvent(PeerScoring scoring, EventType event, PunishmentCalculator calculator) {
-        scoring.recordEvent(event);
-        boolean reputation = scoringCalculator.hasGoodReputation(scoring);
+    private void recordEventAndStartPunishment(PeerScoring peerScoring, EventType event, PunishmentCalculator punishmentCalculator, NodeID nodeID) {
+        peerScoring.recordEvent(event);
+
+        boolean shouldStartPunishment = !scoringCalculator.hasGoodReputation(peerScoring) && peerScoring.hasGoodReputation();
+        if (shouldStartPunishment) {
+            long punishmentTime = punishmentCalculator.calculate(peerScoring.getPunishmentCounter(), peerScoring.getScore());
+            peerScoring.startPunishment(punishmentTime);
+
+            String nodeIDFormated = nodeIdForLog(nodeID);
+            logger.debug("NodeID {} has been punished for {} milliseconds. Reason {}", nodeIDFormated, punishmentTime, event);
+            logger.debug("{}", new PeerScoringInformation(peerScoring, nodeIDFormated, ""));
+        }
+    }
+
+    private String nodeIdForLog(NodeID id) {
+        if(id == null) {
+            return "NO_NODE_ID";
+        }
+        return id.toString().substring(id.toString().length() - 7, id.toString().length() - 1);

```

Comment:
Calculate "toString" only once.

Reply:
Be careful with format, we already unified the way nodeID are printed, i think.

Reply:
changed :)